QHyper/converter.py

```diff
@@ -147,60 +148,32 @@ def to_dimod_qubo(problem: Problem, lagrange_multiplier: float = 10
         return cast(tuple[dict[tuple[str, ...], float], float],
                     bqm.to_qubo())  # (qubo, offset)
 
-    # TODO: Refactor is needed
-    @staticmethod
-    def to_dqm(problem: Problem) -> DiscreteQuadraticModel:
-        if hasattr(problem, "one_hot_encoding"):
-            if problem.one_hot_encoding:
-                return Converter._dqm_from_one_hot_representation(problem)
-            return Converter._dqm_from_discrete_representation(problem)
-        return Converter._dqm_from_one_hot_representation(problem)
 
     @staticmethod
-    def _dqm_from_discrete_representation(problem: Problem) -> DiscreteQuadraticModel:
-        dqm = dimod.DiscreteQuadraticModel()
-
-        for var in problem.variables:
-            var = str(var)
-            if var not in dqm.variables:
-                dqm.add_variable(problem.cases, var)
-
-        for vars, bias in problem.objective_function.as_dict().items():
-            x_i, x_j = vars
-            xi_idx: int = cast(int, dqm.variables.index(x_i))
-            xj_idx: int = cast(int, dqm.variables.index(x_j))
-
-            # We're skipping the linear terms
-            if xi_idx == xj_idx:
-                continue
-
-            dqm.set_quadratic(
-                dqm.variables[xi_idx],
-                dqm.variables[xj_idx],
-                {(case, case): bias for case in range(problem.cases)},
-            )
-
-        return dqm
-
-    @staticmethod
-    def _dqm_from_one_hot_representation(problem: Problem) -> DiscreteQuadraticModel:
-        dqm = dimod.DiscreteQuadraticModel()
-
-        CASES_OFFSET = 1 if problem.cases == 1 else 0
+    def to_dqm(problem: Problem) -> DiscreteQuadraticModel:
```

Comment:
Does this function should only consider objective function and totally skip constraints?

Reply:
Yes, DQM doesn't have ability to add constraints

Reply:
I see, but that doesn't mean that we want to ignore it. As we speak before, without adding constraints you're changing the problem. Imo there should be logic that will convert objective function and constraints into qubo and then into DQM. We can talk about that tomorrow